packages/main/src/TimePicker.ts

```diff
@@ -408,12 +413,17 @@ class TimePicker extends UI5Element implements IFormInputElement {
 		return !isDesktop() && (isPhone() || isTablet());
 	}
 
+	get shouldDisplayValueStateMessageInResponsivePopover() {
+		return this.hasValueStateText && !this._inputsPopover?.open;
+	}
+
 	onTimeSelectionChange(e: CustomEvent<TimeSelectionChangeEventDetail>) {
 		this.tempValue = e.detail.value; // every time the user changes the time selection -> update tempValue
 	}
 
 	_togglePicker() {
 		this.open = !this.open;
+		this._isMobileDevice && (this._inputsPopover!.open = false);
```

Comment:
It's against best practices to use a side effect for assignment as this makes the code harder to read. Please use `if`